src/components/catalog-promo/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import TitleArrowLink from '@components/title-arrow-link';
+import clsx from 'clsx';
+import { useProfile } from '@hooks/use-profile';
+import api from '@services/api';
+import type { Category } from '@services/generated-api/data-contracts';
+import styles from './catalog-promo.module.scss';
+
+const CatalogPromo = () => {
+	const [categories, setCategories] = useState<Category[]>([]);
+	const [numberToShow, setNumberToShow] = useState(6);
+	const { isMobileScreen } = useProfile();
+
+	const findNumbersToShow = useCallback(() => {
+		if (isMobileScreen) {
+			setNumberToShow(6);
+			return;
+		}
+
+		setNumberToShow(5);
+	}, [isMobileScreen]);
+
+	useEffect(() => {
+		findNumbersToShow();
+	}, [findNumbersToShow, isMobileScreen]);
+
+	const handleCategoryToggle = () => {
+		if (numberToShow < 12) {
+			setNumberToShow(12);
+			return;
+		}
+		findNumbersToShow();
+	};
+
+	useEffect(() => {
+		api.categoriesList().then((categories) => setCategories(categories));
+	}, []);
+	return (
+		<>
+			<div className={styles.link}>
+				<TitleArrowLink title="Каталог" link="/catalog/" />
```

Comment:
Чтобы не плодить обертки у компонента "TitleArrowLink" можно добавить props classnames и получить что-то вроде этого:

```suggestion
				<TitleArrowLink title="Каталог" link="/catalog/" classnames={styles.link} />
```

В компоненте "TitleArrowLink" нужно будет доработать логику маленько: 

```suggestion
				const TitleArrowLink: React.FC<TitleArrowLinkProps> = ({ title, link, type, classnames }) => {
	return (
		<Link to={link} className={cn(styles.link, classnames)}>
			<p
				className={clsx(
					styles.link__title,
					type === 'catalogPage' && styles.link__title_type_smallFont
				)}
			>
				{title}
			</p>
			<ArrowIcon
				className={clsx(
					styles.link__arrow,
					type === 'catalogPage' && styles.link__arrow_type_small
				)}
			/>
		</Link>
	);
};
```